### PR TITLE
doc: Bootloader and DFU rework

### PR DESCRIPTION
**Pull request by annwoj**
Based on jira NCSDK27741.
Restructures the DFU and bootloader documentation for the better user experience.
Adds a new quick start guide.

### CODE REVIEW

File: doc/nrf/config_and_build/bootloaders_dfu/suit/suit_dfu.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.. _ug_suit_dfu:
+
+Software Updates for Internet of Things
```
Comment (annwoj) [inline doc/nrf/config_and_build/bootloaders_dfu/suit/suit_dfu.rst:3]: TBC if should be removed for now

File: doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start.rst [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+******************
+
+In case you do not have experience with bootloaders, we recommend going through the following learning resources first:
+
+* `Bootloaders and DFU/FOTA`_ - The page is a part of the `Nordic Developer Academy`_ courses and offers is an introduction to bootloaders, DFU, and various transport features available in the |NCS|.
+* `Adding Device Firmware Update (DFU/FOTA) Support in nRF Connect SDK`_ - The webinar provides an overview of fundamental principles and best practices for adding DFU/FOTA support in an nRF Connect SDK-based firmware.
+
+.. rst-class:: numbered-step
+
+Conceptual documentation
```
Comment (wbober) [inline doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start.rst:25]: Let's name this "Documentation"

In the section add references to mcuboot documentation and our downstream docs.

**Review by wbober — APPROVED** [review]

File: doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start.rst (hunk 2 above)
Comment (annwoj) [reply doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start.rst]: updated, thanks!

**Review by greg-fer — CHANGES_REQUESTED** [review]
Can I ask about the long-term plan for this section? I cannot find this information in the task linked in the description and on the Confluence page linked in that task, and both are marked as WIP.

The reason I'm asking is because #13170 last year separated Bootloader and DFU into two sections based on feedback coming from users @hellesvik-nordic worked with. Many other edits followed suit throughout 2023 to make information more clear for users. Is this coming from similar interaction with users and customers?

**Comment by wbober** [comment]
@greg-fer What change are you exactly requesting? Could you please point to the exact feedback which lead to the conclusion that separating these two topics clearer for users?

**Comment by greg-fer** [comment]
> Could you please point to the exact feedback which lead to the conclusion that separating these two topics clearer for users?

That is based on feedback from @hellesvik-nordic , mentioned in NCSIDB-964, so he is the best to answer.

> @greg-fer What change are you exactly requesting? 

I'm requesting not to merge Bootloader and DFU without a clear long term plan for the section. I don't see anything that is not marked as "WIP" on Confluence, JIRA, or even this PR description.

File: doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_partitioning.rst [hunk 3]
```diff
@@ -1,13 +1,97 @@
...
+For scenarios involving DFU, read the following sections.
+
+.. _ug_bootloader_flash_static_requirement:
+
+Static partition requirement for DFU
+************************************
+
+If you want to perform DFU, you must :ref:`define a static partition map <ug_pm_static>` to ensure compatibility across different builds.
+The dynamically generated partitions can change between builds.
+This is important also when you use a precompiled HEX file as a child image instead of building it.
```
Comment (nvlsianpu) [inline doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_partitioning.rst:21]: child image (sub-image)
rationale: sysbuild uses different terminology.

File: doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start.rst [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+.. list-table:: Bootloaders supported by |NCS|
+   :widths: auto
+   :header-rows: 1
+
+   * - API
+     - Description
+     - Supported transfer
+   * - :ref:`zephyr:mcu_mgr`
+     - Library in Zephyr that implementing the Simple Management Protocol (SMP), which is used to receive or send updates over different protocols.
+     - Wired (SMP over UART or USB) and OTA (SMP over Bluetooth® LE)
```
Comment (nvlsianpu) [inline doc/nrf/config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start.rst:84]: `USB` -> `USB virtual serial port`

**Review by nvlsianpu — APPROVED** [review]
Just a few nits

**Pull request merged**